rust/xaynet-server/src/storage/redis/mod.rs

```diff
@@ -280,48 +273,49 @@ impl Connection {
             .arg(PublicSigningKeyWrite::from(update_pk))
             .invoke_async(&mut self.connection)
             .await
+            .map_err(to_storage_err)
     }
 
-    /// Gets the [`SeedDict`] or an empty [`SeedDict`] when the [`SumDict`] does not exist.
-    ///
     /// # Note
     /// This method is **not** an atomic operation.
-    pub async fn get_seed_dict(mut self) -> RedisResult<SeedDict> {
+    async fn seed_dict(&mut self) -> StorageResult<Option<SeedDict>> {
         debug!("get seed dictionary");
         // https://redis.io/commands/hkeys
         // > Return value:
         //   Array reply: list of fields in the hash, or an empty list when key does not exist.
         let sum_pks: Vec<PublicSigningKeyRead> = self.connection.hkeys("sum_dict").await?;
 
-        let mut seed_dict: SeedDict = SeedDict::new();
-        for sum_pk in sum_pks {
-            // https://redis.io/commands/hgetall
-            // > Return value
-            //   Array reply: list of fields and their values stored in the hash, or an empty
-            //   list when key does not exist.
-            let sum_pk_seed_dict: HashMap<PublicSigningKeyRead, EncryptedMaskSeedRead> =
-                self.connection.hgetall(&sum_pk).await?;
-            seed_dict.insert(
-                sum_pk.into(),
-                sum_pk_seed_dict
-                    .into_iter()
-                    .map(|(pk, seed)| (pk.into(), seed.into()))
-                    .collect(),
-            );
-        }
-
-        Ok(seed_dict)
+        let result = if sum_pks.is_empty() {
```

Comment:
Like above, I think I prefer a version with an early return in case `sum_pks` is empty.

Reply:
fixed